### PR TITLE
#865 Collection NPE check for a calling update method.

### PR DESCRIPTION
**Pull request by sjaakd**

### CODE REVIEW

File: processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java [hunk 1]
```diff
@@ -447,14 +448,25 @@ private Assignment assignToCollection(Type targetType,
...
+                        // if the calling method is an update method and accesses a getter, make a local variable to
+                        // test NPE first.
+                        result = new LocalVarWrapper( result, method.getThrownTypes(), targetType );
+                    }
+                    else {
+                        // if not, asssign as new collecitin or direct
+                        result = new SetterWrapper( result, method.getThrownTypes() );
+                    }
+
+                    // TODO.. Reviewer, why is there a special collectionGetterName?
```
Comment (agudian) [inline processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java]: Hmm, it turns the setter method to the name of the getter. That might actually make sense, but I don't think we're checking if it actually exists. Hmm.

File: processor/src/test/java/org/mapstruct/ap/test/bugs/_865/ProjectEntity.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    public List<ProjCoreUserEntity> getCoreUsers() {
+        return coreUsers;
+    }
+
+    public void setCoreUsers(List<ProjCoreUserEntity> coreUsers) {
+        this.coreUsers = coreUsers;
+    }
+
+
+
```
Comment (agudian) [inline processor/src/test/java/org/mapstruct/ap/test/bugs/_865/ProjectEntity.java]: Three newlines... :)

**Comment by agudian** [comment]
Awesome, @sjaakd! Looks very good :+1:!

File: processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java (hunk 1 above)
Comment (sjaakd) [inline processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java]: > Hmm, it turns the setter method to the name of the getter. That might actually make sense, but I don't think we're checking if it actually exists. Hmm.

But we should be able to derive this in an alternative way as well, right? We have the `Type#Map<String, ExecutableElement> getPropertyReadAccessors()`

File: processor/src/test/java/org/mapstruct/ap/test/bugs/_865/ProjectEntity.java (hunk 2 above)
Comment (sjaakd) [inline processor/src/test/java/org/mapstruct/ap/test/bugs/_865/ProjectEntity.java]: They're free :)..

**Comment by sjaakd** [comment]
> Awesome, @sjaakd! Looks very good 👍!

Thanks.. I just saw there are remainders of other these (the no getter test) as well. I'll add a testcase for that or remove it.

Then I'll merge and push.

File: processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java (hunk 1 above)
Comment (sjaakd) [inline processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java]: I'll make an issue out of it to check if its really needed. See #878

**Comment by sjaakd** [comment]
please test this

**Pull request merged**